### PR TITLE
Categories

### PR DESCRIPTION
**Pull request by kenz90**
completed

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const seed = require("../db/seeds/seed");
+const testData = require("../db/data/test-data");
+const db = require("../db/connection");
+const app = require("../app");
+const request = require("supertest");
+
+beforeEach(() => seed(testData));
+afterAll(() => {
+  db.end();
```
Comment (pkaodev) [inline __tests__/app.test.js]: since you have block quoted db.end() you want to return it here (it is a promise)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+});
+
+describe("/api/categories", () => {
+  describe("GET /api/categories", () => {
+    test("status: 200, should respond with all treasures", () => {
```
Comment (pkaodev) [inline __tests__/app.test.js]: treasures?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      return request(app)
+        .get("/api/categories")
+        .expect(200)
+        .then((result) => {
+          expect(result.body.categories).toHaveLength(4);
+          result.body.categories.forEach((category) => {
+            expect(category).toEqual(
```
Comment (pkaodev) [inline __tests__/app.test.js:21]: have a look at lines 21 and 22, we don't need 2 expect statements here (just one for each category).

File: __tests__/app.test.js (hunk 2 above)
Comment (kenz90) [reply __tests__/app.test.js]: lol, should be categories

**Review by pkaodev — CHANGES_REQUESTED** [review]
Looking good, just a couple of things need tweaking (have a read of all the comments first):
-update the small changes
-fix the expect statement
-what happens if you send a request to /api/caaaaategories (or any other typo?)  Let's make a test for this, and an error handler.

after you've made these changes send another nchelp PR please

File: __tests__/app.test.js (hunk 3 above)
Comment (kenz90) [reply __tests__/app.test.js:21]: I've tried getting rid of expect but it doesn't seem to work without it

Comment (Poonam-raj) [reply __tests__/app.test.js:21]: All good in that case

File: __tests__/app.test.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    });
+  });
+  describe("GET /api/categories ERRORS", () => {
+    test("status: 404, should respond with error message not found when given incorrect route", () => {
+      return request(app)
+        .get("/api/caaateoriesus")
+        .expect(404)
+        .then((res) => {
+          expect(res.body.msg).toBe("not found");
+        });
```
Comment (Poonam-raj) [inline __tests__/app.test.js:38]: nice error test - this test block could slot nicely into the describe block above as it is to do with the same endpoint but just another status outcome

**Review by Poonam-raj — APPROVED** [review]
All good to merge!

**Pull request merged**